core/src/processing/opengl/PSurfaceJOGL.java

```diff
@@ -273,6 +273,15 @@ protected void initGL() {
       } else if (PJOGL.profile == 2) {
         try {
           profile = GLProfile.getGL2ES2();
+
+          // workaround for https://jogamp.org/bugzilla/show_bug.cgi?id=1347
+          if (!profile.isHardwareRasterizer()) {
+            GLProfile hardware = GLProfile.getMaxProgrammable(true);
```

Comment:
if i am not mistaken you should use
GLProfile.getMaxFixedFunc(true);
here.
see: https://jogamp.org/bugzilla/show_bug.cgi?id=1347#c2